README.md

```diff
@@ -53,7 +53,42 @@ cd onyo
 pip install -e ".[tests, docs]"
 ```
 
-### Tests
+### Design
+
+Onyo follows a set of development and coding conventions throughout its code
+base. For linting and type checking, the Onyo project uses flake8 and Pyre.
+
+#### the complete code base
+- All classes, functions and properties have a doc-string describing their
+  functionality.
+- All functions have type hinting (including explicit return type for `None`).
+- The code is PEP8 conform.
+
+#### lib Tests (`onyo/lib/`)
+- they follow Onyos code conventions, see `"the complete code base"
+- The tests proof stability of the repository class and the rest of the code to
+  offer a stable base for the consumers of their code (i.e. CLI, demos)
+- all public functions of all classes are tested:
+  - for all possible existing input arguments
+    - single valid inputs used
+    - multiple valid inputs used (if allowed)
+    - valid test without input (if allowed)
+    - one test for each possible error
+    - one test for each possible conflict with other inputs
+
+#### CLI Tests (`tests/`)
+- they follow Onyos code conventions, see `"the complete code base"
+- The tests define Onyos behavior the user can expect, and they are created
```

Comment:
As a style suggestion, to reduce redundancy, each bullet point can start with a verb:

For example:

CLI tests:
- follow [...]
- define [...]
- etc